### ISSUE TITLE
[Bug]: TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'

### ISSUE THREAD

**Issue by comet424** [firstof9/ha-gasbuddy#115]
### The problem

Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:599
integration: Sensor ([documentation](https://www.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
First occurred: 1:17:09 PM (4 occurrences)
Last logged: 1:17:09 PM

Error adding entity sensor.neejees_premium_gas for domain sensor with platform gasbuddy
Error adding entity sensor.esso_hastings_midgrade_gas for domain sensor with platform gasbuddy
Error adding entity sensor.esso_hastings_premium_gas for domain sensor with platform gasbuddy
Error adding entity sensor.esso_hastings_diesel for domain sensor with platform gasbuddy
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 599, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 918, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1368, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1007, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1132, in _async_write_ha_state
    self.__async_calculate_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1069, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1013, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/sensor/__init__.py", line 544, in state
    value = self.native_value
            ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/gasbuddy/sensor.py", line 89, in native_value
    self._state = data[self._type]["price"] / 100
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~
TypeError: unsupported operand type(s) for /: 'NoneType' and 'int'

### What version of Home Assistant?

2024.11.3

### What version of GasBuddy?

1.0.16

### Diagnostics information

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

**Comment by firstof9** [comment]
This should be fixed in [1.0.17-b1](https://github.com/firstof9/ha-gasbuddy/releases/tag/1.0.17-b1)
Please give it a try and let me know if you're fixed.

**Comment by comet424** [comment]
so far so good  no errors yet.. well hopefully no errors later  but so far  no issues ill see what 24 hours brings but looks good so far
